cypress/e2e/shared/dashboardsView.test.ts

```diff
@@ -1392,4 +1392,268 @@ csv.from(csv: data) |> filter(fn: (r) => r.bucket == v.bucketsCSV)`
+      cy.intercept('POST', '/query', req => {
+        req.alias = 'refreshQuery'
+      })
+      cy.getByTestID('auto-refresh-overlay').within(() => {
+        cy.getByTestID('refresh-form-activate-button').click()
+      })
+      cy.wait('@refreshQuery')
+      cy.wait('@refreshQuery')
+      cy.getByTestID('enable-auto-refresh-button').click()
+      cy.getByTestID('enable-auto-refresh-button').then(el => {
+        expect(el[0].getAttribute('title')).to.equal('Enable Auto Refresh')
+      })
+      cy.wait('@refreshQuery')
+      cy.on('fail', err => {
+        expect(err.message).to.equal(
+          'Timed out retrying after 5000ms: `cy.wait()` timed out waiting `5000ms` for the 3rd request to the route: `refreshQuery`. No request ever occurred.'
+        )
+        done()
+      })
+    })
+
+    it('can timeout on a preset timeout selected by the user', () => {
+      cy.getByTestID('enable-auto-refresh-button').click()
+      cy.getByTestID('auto-refresh-overlay').should('be.visible')
+      cy.getByTestID('auto-refresh-overlay').within(() => {
+        cy.getByTestID('autorefresh-dropdown--button').click()
+        cy.getByTestID('auto-refresh-5s').click()
+        cy.getByTestID('timerange-dropdown').click()
+      })
+      cy.getByTestID('timerange-popover--dialog').within(() => {
+        cy.getByTestID('timerange--input')
+          .clear()
+          .type(`${jumpAheadTime('00:00:11')}`)
+        cy.getByTestID('daterange--apply-btn').click()
+      })
+      cy.intercept('POST', '/query', req => {
+        req.alias = 'refreshQuery'
+      })
+
+      cy.getByTestID('auto-refresh-overlay').within(() => {
+        cy.getByTestID('refresh-form-activate-button').click()
+      })
+
+      cy.wait('@refreshQuery')
+      cy.wait('@refreshQuery')
+
+      cy.wait(2000)
+
+      cy.getByTestID('enable-auto-refresh-button').then(el => {
+        expect(el[0].getAttribute('title')).to.equal('Enable Auto Refresh')
+      })
+    })
+
+    it('does not refresh if user leaves, until user comes back, and then continues', () => {
+      cy.getByTestID('enable-auto-refresh-button').click()
+      cy.getByTestID('auto-refresh-overlay').should('be.visible')
+      cy.getByTestID('auto-refresh-overlay').within(() => {
+        cy.getByTestID('autorefresh-dropdown--button').click()
+        cy.getByTestID('auto-refresh-5s').click()
+        cy.getByTestID('timerange-dropdown').click()
+      })
+      cy.getByTestID('timerange-popover--dialog').within(() => {
+        cy.getByTestID('timerange--input')
+          .clear()
+          .type(`${jumpAheadTime('00:00:10')}`)
+        cy.getByTestID('daterange--apply-btn').click()
+      })
+      cy.intercept('POST', '/query', req => {
+        req.alias = 'refreshQuery'
+      })
+
+      cy.getByTestID('auto-refresh-overlay').within(() => {
+        cy.getByTestID('refresh-form-activate-button').click()
+      })
+
+      cy.wait('@refreshQuery')
+
+      cy.visit('/')
+
+      cy.wait(5000)
```

Comment:
why?

Reply:
because we want to make sure that within the timeframe of a refresh interval, we aren't refreshing despite the auto refresh being on.

Reply:
k